### PR TITLE
Added cancelation feature

### PR DESCRIPTION
**Pull request by MichalPavlik**
PR to the feature branch. It enables cancellation of the running build.

### CODE REVIEW

**Comment by danmoseley** [comment]
Should you have a test? It can just sleep into canceled, or sleep for 30 sec and if not canceled by then, Fail the test.

File: src/Build/BackEnd/Node/OutOfProcServerNode.cs [hunk 1]
```diff
@@ -269,11 +273,40 @@ private void HandlePacket(INodePacket packet)
...
+                    break;
+                case NodePacketType.ServerNodeBuildCancel:
+                    HandleServerNodeBuildCancel((ServerNodeBuildCancel)packet);
                     break;
             }
         }
 
+        private void HandleServerNodeBuildCancel(ServerNodeBuildCancel command)
+        {
+            BuildManager.DefaultBuildManager.CancelAllSubmissions();
```
Comment (AR-May) [inline src/Build/BackEnd/Node/OutOfProcServerNode.cs:286]: I wonder, why can't we reuse here the cancellation from the MSBuildApp.Execute() function, by throwing the cancellation event manually? Also, how the things about the server after the current cancellation implementation works? Does server send the ServerNodeBuildResult package with the relevant information? A unit test would be helpful here, as we would see the intended output after the cancellation and the fact that it was finalized.

Comment (MichalPavlik) [reply src/Build/BackEnd/Node/OutOfProcServerNode.cs:286]: We discussed with Roman what we can reuse from the cancellation code in XMake (Console_CancelKeyPress). It contains some handling for interactive console, but server node works in non-interactive console.

Comment (MichalPavlik) [reply src/Build/BackEnd/Node/OutOfProcServerNode.cs:286]: I planned to add test when Nathan's PR with server node tests is merged. We can merge his PR first and I will add test for cancelation.

**Review by AR-May — APPROVED** [review]
Feel free to merge it to the feature branch, and implement the test in other PR, as discussed!
I would be able to work on client side cancellation as soon as this PR is merged.

**Review by AR-May — APPROVED** [review]

**Pull request merged**